game-core/src/main/java/games/strategy/engine/chat/MessengersChatTransmitter.java

```diff
@@ -4,38 +4,39 @@
 import games.strategy.engine.message.MessageContext;
 import games.strategy.engine.message.RemoteName;
 import games.strategy.net.Messengers;
-import java.util.Map;
+import java.util.Collection;
 
 /** Chat transmitter that sends and receives messages over Java NIO sockets. */
-class JavaSocketChatTransmitter implements ChatTransmitter {
-  private final PlayerName playerName;
+public class MessengersChatTransmitter implements ChatTransmitter {
   private final Messengers messengers;
 
-  private final IChatChannel chatChannelSubscriber;
+  private IChatChannel chatChannelSubscriber;
 
   private final String chatName;
   private final String chatChannelName;
 
-  JavaSocketChatTransmitter(
-      final ChatClient chatClient, final String chatName, final Messengers messengers) {
-    this.playerName = messengers.getLocalNode().getPlayerName();
+  public MessengersChatTransmitter(final String chatName, final Messengers messengers) {
     this.messengers = messengers;
     this.chatName = chatName;
     this.chatChannelName = ChatController.getChatChannelName(chatName);
+  }
+
+  @Override
+  public void setChatClient(final ChatClient chatClient) {
     chatChannelSubscriber = chatChannelSubscriber(chatClient);
   }
 
   private IChatChannel chatChannelSubscriber(final ChatClient chatClient) {
     return new IChatChannel() {
       @Override
       public void chatOccurred(final String message) {
-        chatClient.messageReceived(message);
+        chatClient.messageReceived(MessageContext.getSender().getPlayerName(), message);
```

Comment:
I still don't like the `MessageContext` class :/

Reply:
Totally, please take note where this line is being removed [from in Chat.java](https://github.com/triplea-game/triplea/pull/5322/files#diff-8ffdcb54556ca7b66cd59050e599e646L66)

That blew up when switching out the chat transmitter type.

Reply:
Looking forward to the day where this class can be safely removed ^^